packages/frontend/src/components/composer/Composer.tsx

```diff
@@ -743,10 +915,149 @@ export function useDraft(
 
   return {
     draftState,
+    onSelectReplyToShortcut,
     removeQuote,
     updateDraftText,
     addFileToDraft,
     removeFile,
     clearDraft,
   }
 }
+
+function useMessageEditing(
+  accountId: number,
+  chatId: T.BasicChat['id'],
+  editMessageInputRef: React.MutableRefObject<ComposerMessageInput | null>
+) {
+  const [_originalMessage, setOriginalMessage] = useState<null | T.Message>(
+    null
+  )
+  // We unset `_originalMessage` when switching between chats below,
+  // but let's still make sure that it belongs to the current chat here.
+  const originalMessage =
+    _originalMessage?.chatId === chatId ? _originalMessage : null
+
+  // Should we also listen for `MsgsChanged` and update the message?
+  // E.g. if it got edited from another device.
+
+  const isEditingModeActive = originalMessage != null
+
+  const cancelEditing = useCallback(() => {
+    setOriginalMessage(null)
+    // This should be unnecessary because `setOriginalMessage(null)`
+    // should be enough, but let's sill clean up.
+    editMessageInputRef.current?.setText('')
+  }, [editMessageInputRef])
+
+  if (useHasChanged2(chatId)) {
+    // Yes, this means that the "edit draft" gets lost on chat change.
+    // Though maybe we can easily keep it between chat switches,
+    // at least for one chat?
+    // Because we do check if `_originalMessage` belongs to the current chat,
```

Comment:
Since the implementation allows it, I would go for it. Only counter argument: users won't "understand" why the edit draft sometimes is preserved and sometimes not.